drivers/adc/adc_mchp_g1.c

```diff
@@ -0,0 +1,917 @@
+/*
+ * Copyright (c) 2025 Microchip Technology Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <soc.h>
+#include <zephyr/drivers/adc.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/drivers/clock_control/mchp_clock_control.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/irq.h>
+#define ADC_CONTEXT_USES_KERNEL_TIMER
+#include "adc_context.h"
+
+LOG_MODULE_REGISTER(adc_mchp_g1, CONFIG_ADC_LOG_LEVEL);
+
+#define DT_DRV_COMPAT microchip_adc_g1
+
+#define ADC_REGS         ((const struct adc_mchp_dev_config *)(dev)->config)->regs
+#define ADC_MCHP_VREF_MV DT_INST_PROP(0, vref_mv)
+
+/* Calculate ADC sample length (SAMPLEN) from sampling time (ns), ADC GCLK (Hz),
+ * and prescaler value.
+ *
+ * SAMPLEN = ((sampling_time_ns * gclk_adc_hz) / (prescaler_val * 1_000_000_000)) - 1
+ */
+#define ADC_CALC_SAMPLEN_NS(sampling_time_ns, gclk_adc_hz, prescaler_val)                          \
+	((((uint64_t)(sampling_time_ns) * (gclk_adc_hz)) / ((prescaler_val) * 1000000000ULL)) - 1)
+
+/* Macro to Write the calibration values */
+#define ADC_CALIB_WRITE_FIELD(adc_id, field)                                                       \
+	do {                                                                                       \
+		volatile uint32_t *sw0_word = (volatile uint32_t *)ADC_CALIBRATION_AREA;           \
+		uint8_t val = (((*sw0_word) & ADC##adc_id##_##field##_Msk) >>                      \
+			       ADC##adc_id##_##field##_Pos);                                       \
+		ADC_REGS->ADC_CALIB |= ADC_CALIB_##field(val);                                     \
+	} while (0)
```

Comment:
Is the difference between one instance to the another just an offset and the positions are the same? If yes, you should add the offset in devicetree instead to use adc_id.